libs/ui-watt/src/lib/components/shell/nav-list/watt-nav-list.component.ts

```diff
@@ -15,28 +15,82 @@
  * limitations under the License.
  */
 
+import { CommonModule } from '@angular/common';
 import {
   ChangeDetectionStrategy,
   Component,
+  ContentChildren,
+  HostBinding,
+  Input,
   NgModule,
+  QueryList,
   ViewEncapsulation,
 } from '@angular/core';
-
+import { RouterModule } from '@angular/router';
+import { MatExpansionModule } from '@angular/material/expansion';
 import { MatListModule } from '@angular/material/list';
-import { WattNavListItemScam } from './watt-nav-list-item.component';
+
+import { WattExpandOnActiveLinkDirective } from './watt-expand-on-active-link.directive';
+import {
+  WattNavListItemComponent,
+  WattNavListItemScam,
+} from './watt-nav-list-item.component';
 
 @Component({
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
   selector: 'watt-nav-list',
   styleUrls: ['./watt-nav-list.component.scss'],
-  template: `<mat-nav-list><ng-content></ng-content></mat-nav-list>`,
+  template: `
+    <ng-container *ngIf="expandable; else navListTemplate">
+      <mat-expansion-panel
+        wattExpandOnActiveLink
+        [wattNavListItemComponents]="navListItemComponents"
+        class="mat-elevation-z0"
+      >
+        <mat-expansion-panel-header>
+          <mat-panel-title class="watt-text-m">{{ title }}</mat-panel-title>
+        </mat-expansion-panel-header>
+        <ng-container *ngTemplateOutlet="navListTemplate"></ng-container>
+      </mat-expansion-panel>
+    </ng-container>
+
+    <ng-template #navListTemplate>
+      <mat-nav-list><ng-content></ng-content></mat-nav-list>
+    </ng-template>
+  `,
```

Comment:
Question: Is it time to move the template to a separate file? What is our preference on this?

Reply:
I don't think there's a clear guideline for when an inline template should be moved to a separate file. Probably a good idea in this case. Have done it 👍